lib/psmsl.py

```diff
@@ -0,0 +1,104 @@
+import io
+import datetime
+
+import numpy as np
+import pandas as pd
+
+
+def missing2nan(value, missing=-99999):
+    """convert the value to nan if the float of value equals the missing value"""
+    value = float(value)
+    if value == missing:
+        return np.nan
+    return value
+
+def year2date(year_fraction, dtype='datetime64[s]'):
+    """convert a fraction of a year + fraction of a year to a date, for example 1993.12 -> 1993-02-01.
```

Comment:
fraction of a year + fraction of a year -> wut?

Reply:
```suggestion
    """convert a fraction of a year + fraction of a year to a date, for example 1993.083 -~> 1993-02-01.
```